lib/process/process_test.go

```diff
@@ -324,3 +326,162 @@ type importerSuite struct {
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "false",
+			},
+			needUpdate: false,
+		},
+		{
+			name: "reconciliation mode = enabled. Different labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "true",
+					"label1":                     "1",
+					"label2":                     "1",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "true",
+				"label1":                     "value1",
+				"label2":                     "value2",
+			},
+			needUpdate: true,
+		},
+		{
+			name: "reconciliation mode = enabled. Same labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "true",
+					"label1":                     "value1",
+					"label2":                     "value2",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "true",
+				"label1":                     "value1",
+				"label2":                     "value2",
+			},
+			needUpdate: false,
+		},
+		{
+			name: "reconciliation mode = EnsureExists. Different labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "EnsureExists",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "EnsureExists",
+				"label1":                     "value1",
+				"label2":                     "value2",
+			},
+			needUpdate: true,
+		},
+		{
+			name: "reconciliation mode = EnsureExists. Different value of labels",
+			args: args{
+				currentLabels: map[string]string{
+					"gravitational.io/reconcile": "EnsureExists",
+					"label1":                     "1",
+					"label2":                     "2",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "EnsureExists",
+				"label1":                     "1",
+				"label2":                     "2",
+			},
+			needUpdate: false,
+		},
+		{
+			name: "reconciliation mode is empty. Same labels",
+			args: args{
+				currentLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+				requiredLabels: map[string]string{
+					"label1": "value1",
+					"label2": "value2",
+				},
+			},
+			expectedLabels: map[string]string{
+				"gravitational.io/reconcile": "EnsureExists",
+				"label1":                     "value1",
+				"label2":                     "value2",
+			},
+			needUpdate: true,
+		},
+		{
+			name: "reconciliation mode is incorrect. Different value of labels",
```

Comment:
ditto here:
```
name: "reconciliation mode is incorrect. Different value of labels and no change is expected",
```
Consider adding a test that an invalid mode does not behave as `EnsureExists` (e.g. will not add labels)

Reply:
Sure